pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java

```diff
@@ -446,8 +447,11 @@ private Object applyAnswerPredicate(InterpreterContext ictx, String userGuid, St
         if (ictx.getActivityInstanceSummary() != null) {
             questionType = ictx.getActivityInstanceSummary()
                     .getLatestActivityInstance(activityCode)
-                    .flatMap(instanceDto -> ActivityDefStore.getInstance()
-                            .findActivityDef(ictx.getHandle(), studyGuid, instanceDto))
+                    .map(instanceDto -> ActivityInstanceUtil.getActivityDef(
```

Comment:
`ActivityDefStore::findActivityDef` attempts to look for definition in the cache, but DOES NOT load it if it is missing. `ActivityInstanceUtil::getActivityDef` will do the right thing of loading it if there's a cache miss.

Reply:
May be in the long those public methods in ActivityDefStore should try to load if not found in cache else we might run into these sort of issues if invoked immediately after AppEngine restart.